modules/4337/test/erc4337/Safe4337Module.spec.ts

```diff
@@ -268,6 +268,40 @@ describe('Safe4337Module', () => {
 
       expect(await safeFromEntryPoint.validateUserOp.staticCall(userOp, ethers.ZeroHash, 0)).to.eq(packedValidationData)
     })
+
+    it('should indicate failed validation data when signature length contains additional bytes', async () => {
+      const { user, safeModule, validator, entryPoint } = await setupTests()
+
+      const validAfter = BigInt(ethers.hexlify(ethers.randomBytes(3)))
+      const validUntil = validAfter + BigInt(ethers.hexlify(ethers.randomBytes(3)))
+
+      const safeOp = buildSafeUserOpTransaction(
+        await safeModule.getAddress(),
+        user.address,
+        0,
+        '0x',
+        '0',
+        await entryPoint.getAddress(),
+        false,
+        false,
+        {
+          validAfter,
+          validUntil,
+        },
+      )
+
+      const safeOpHash = calculateSafeOperationHash(await validator.getAddress(), safeOp, await chainId())
+      const signature = buildSignatureBytes([await signHash(user, safeOpHash)]).concat('00')
+      const userOp = buildPackedUserOperationFromSafeUserOperation({
+        safeOp,
+        signature,
+      })
+      const packedValidationData = packValidationData(1, validUntil, validAfter)
+      const entryPointImpersonator = await ethers.getSigner(await entryPoint.getAddress())
+      const safeFromEntryPoint = safeModule.connect(entryPointImpersonator)
+
+      expect(await safeFromEntryPoint.validateUserOp.staticCall(userOp, ethers.ZeroHash, 0)).to.eq(packedValidationData)
+    })
```

Comment:
You already added tests to the `ReferenceEntryPoint.spec.ts`, so no need to re-add them here, but my personal opinion is that getting the full branch coverage makes more sense with the unit tests here, and not in the above "light integration tests" from the `ReferenceEntryPoint.spec.ts` (and avoids the extra long test names because of the "Note: ...").

Additionally, I still think a test is missing - in particular, the new test with the  invalid dynamic part pointer would have failed _before_ this PR. While the invalid dynamic part pointer is nice to have, I think we also need to add a test with a **valid** Safe signature that only fails the `_checkSignatureLength` function.

Reply:
Added tests here: https://github.com/safe-global/safe-modules/pull/453/commits/c54c89d722c732bee2da399ea7e15e8dd572ef14

I think having tests in both `ReferenceEntryPoint.spec.ts` and `SafeModule.spec.ts` would be good to have because `SafeModule.spec.ts` uses mock safe instance and `ReferenceEntryPoint.spec.ts` uses actual Safe instance. Signature verification flow for 4337 module being critical I think we should also have tests with actual Safe instance.

Reply:
I agree that we _should_ have a test in `ReferenceEntryPoint.spec.ts`, but since this is more of an "integration test light", I don't think we need to necessarily fully cover all paths in all files - and it felt a little strange to me that in the "unit test" file, we have the least amount of coverage, while in the "integration light test" files, we have the fuller coverage.

The tests are there, definitely don't take them out. Just something I wanted to point out.